tests/MongoDB.Driver.Core.Tests/Core/Operations/AsyncCursorTests.cs

```diff
@@ -437,88 +438,22 @@ private bool GetDisposed(IChannelSource channelSource)
 
     public static class AsyncCursorReflector
```

Comment:
After this refactoring all the repetitive code is in the general purpose Reflector class.

The AsyncCursorReflector class is just glue code allowing the use of:

    subject._batchSize()

instead of:

    (int?)Reflector.GetFieldValue(subject, "_batchSize")